### ISSUE TITLE
[.NET 8] Some Symbol Entries in perf-%pid.map Are Truncated

### ISSUE THREAD

**Issue by brianrob** [dotnet/runtime#102938]
### Description

There is a regression in .NET 8 where symbols written to `perf-%pid.map` can be truncated.  This means that profiling tools can display the wrong symbols after a profile has been captured on Linux.

### Reproduction Steps

1. Build the following application against .NET 8 on Linux:

```
using System;

public static class Program
{
    public static void Main(string[] args)
    {
        while(true)
        {
            Console.WriteLine();
        }
    }
}
```

2. Run `export COMPlus_PerfMapEnabled=1` to enable generation of `perf-%pid.map` files.
3. Clear all `perf-%pid.map` files from `/tmp`.
4. Run the application built in step 1 and then kill it via `CTRL+C`.
5. Open `/tmp/perf-%pid.map` for the application.  There is likely only one because we cleared out all the previous files.
6. Search for `ConsolePal::Writ(`.

### Expected behavior

An entry for `System.ConsolePal::Write` should exist.  Here's an example correct entry:

```
0x7fb99c69d090 22a void [System.Console] System.ConsolePal::Write(class [System.Runtime]Microsoft.Win32.SafeHandles.SafeFileHandle,valuetype [System.Runtime]System.ReadOnlySpan`1<uint8>,bool)[MinOptJitted]
```

### Actual behavior

An entry for `System.ConsolePal::Writ` exists, even though there is no such method.  Here's an example entry:

```
0x7fb99c69d090 22a void [System.Console] System.ConsolePal::Writ(class [System.Runtime]Microsoft.Win32.SafeHandles.SafeFileHandle,valuetype [System.Runtime]System.ReadOnlySpan`1<uint8>,bool)[MinOptJitted]
```

### Regression?

Yes.  This is a regression from https://github.com/dotnet/runtime/pull/81243.

This was subsequently fixed in .NET 9 via https://github.com/dotnet/runtime/pull/98687.  I have confirmed that applying this PR to .NET 8 fixes the problem.  Requesting that we backport this change to .NET 8.

### Known Workarounds

None.

### Configuration

 - All versions of .NET 8.
 - All architectures.
 - Linux only.

**Comment by brianrob** [comment]
@jakobbotsch, @janvorli what do you think about porting #98687 to .NET 8?

**Comment by janvorli** [comment]
I think it is trivial enough to be safe to backport.

**Comment by brianrob** [comment]
Thanks @janvorli.  Created backport PR https://github.com/dotnet/runtime/pull/103003.

**Comment by brianrob** [comment]
Fixed via #103003.
